torch/csrc/jit/ir/ir.cpp

```diff
@@ -1206,7 +1206,7 @@ bool Node::hasSideEffects() const {
     return false;
   }
 
-  if (kind_.is_prim() || kind_.is_aten() || kind_.is_cuda()) {
+  if (kind_.is_prim() || kind_.is_aten() || kind_.is_cuda() || kind_.is_xpu()) {
```

Comment:
I'd want to get someone more familiar with torchscript/JIT to take a look at this - is this a reasonable way for other backends to hook into some of torchscript's capabilities? There are also a few other changes related to JIT in this PR.

cc @davidberard98 (or @eellison ?)